tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/ui/pin/PinUnlockActivity.kt

```diff
@@ -70,12 +72,20 @@ class PinUnlockActivity : BasePinActivity() {
     override fun onPinChange(pinLength: Int, intermediatePin: String?) {
         super.onPinChange(pinLength, intermediatePin)
         pinTopText.setTextColor(ContextCompat.getColor(this, R.color.wa_white))
-        pinTopText.text =  getString(
-            when (returnActivity) {
-                ReturnActivity.SETTINGS.getActivityOrder() -> R.string.LockPinSet_Settings_EnterCurrentPin
-                ReturnActivity.CAMOUFLAGE.getActivityOrder() -> R.string.LockPinSet_Settings_EnterCurrentPinToChangeCamouflage
-                else -> R.string.UnlockPin_Message_EnterPin
-            })
+        when (returnActivity) {
+            ReturnActivity.SETTINGS.getActivityOrder() -> {
+                pinTopText.text = getString(R.string.LockPinSet_Settings_EnterCurrentPin)
+                pinEditText.setHint(R.string.LockPinSet_Settings_EnterCurrentPin)
+            }
+            ReturnActivity.CAMOUFLAGE.getActivityOrder() -> {
+                pinTopText.text = getString(R.string.LockPinSet_Settings_EnterCurrentPinToChangeCamouflage)
+                pinEditText.setHint(R.string.LockPinSet_Settings_EnterCurrentPinToChangeCamouflage)
+            }
+            else -> {
+                pinTopText.text = getString(R.string.UnlockPin_Message_EnterPin)
+                pinEditText.setHint(R.string.UnlockPin_Message_EnterPin)
+            }
+        }
```

Comment:
Maybe Instead of calling getString() every time, you could put the string resource IDs in variables and then reuse them.
And then you can use setText() to avoid multiple calls to getText() internally.